site/docs/samples/notebook.md

```diff
@@ -1,162 +1,250 @@
-    }
-    ...
-    ``` 
-
-    - Edit the `client = fl.connect(...)` command to point to the right service and namespace of the arrow-flight-module.
-    - To find the service and namespace run:
+    1. Ensure that [Kubeflow](https://www.kubeflow.org/) is installed in your cluster
+    1. Create a port-forward to communicate with Kubeflow:
         ```bash
-        kubectl get svc -l app.kubernetes.io/name=arrow-flight-module --all-namespaces
+        kubectl port-forward svc/istio-ingressgateway -n istio-system 8080:80 &
         ```
-
-    - Edit `"asset": "<bucket-name>/<file-name>.csv"` in the second command to point to your bucket and the name of the dataset.
-
-1. Run the notebook
-
-    You should observe in the cell `Get Data` the data from the dataset.
-
-1. Finally, kill the port-forward
+    1. Open your browser and go to [http://localhost:8080/](http://localhost:8080/).
+    1. Click **Start Setup** and then **Finish** (use the `anonymous` namespace).
+    1. Click **Notebook Servers** (in the left).
+    1. In the notebooks page select in the top left the `anonymous` namespace and then click **New Server**.
+    1. In the notebook server creation page, set `my-notebook` in the **Name** box and then click **Launch**. Wait for the server to become ready.
+    1. Click **Connect** and create a new notebook in the server.
+
+
+## Create a `M4DApplication` resource for the notebook
+
+Create a [`M4DApplication`](../reference/crds.md#m4dapplication) resource to register the notebook workload to the control plane of Mesh for Data: 
+
+<!-- TODO: role field removed but code still requires it -->
+```yaml
+cat <<EOF | kubectl apply -f -
+apiVersion: app.m4d.ibm.com/v1alpha1
+kind: M4DApplication
+metadata:
+  name: my-notebook
+  labels:
+    app: my-notebook
+spec:
+  selector:
+    workloadSelector:
+      matchLabels:
+        app: my-notebook
+  appInfo:
+    intent: fraud-detection
+  data:
+    - dataSetID: "m4d-notebook-sample/data-csv"
+      requirements:
+        interface: 
+          protocol: m4d-arrow-flight
+          dataformat: arrow
+EOF
+```
+
+Notice that:
+
+* The `selector` field matches the labels of our Jupyter notebook workload.
+* The `data` field includes a `dataSetID` that matches the asset identifier in the catalog.
+* The `protocol` and `dataformat` indicate that the developer wants to consume the data using Apache Arrow Flight.
+
+Wait until the `M4DApplication` is ready:
+```bash
+while [[ $(kubectl get m4dapplication my-notebook -o 'jsonpath={.status.ready}') != "true" ]]; do echo "waiting" && sleep 1; done
+```
+
+## Read the dataset from the notebook
+
+In your notebook insert and run a cell to install pandas and pyarrow packages:
+
+```python
+%pip install pandas pyarrow
+```
+
+Then insert a new cell to read the data. The code to use is available as part of the `M4DApplication` and can be printed from your terminal with:
```

Comment:
remove: Then insert a new cell to read the data. The code to use is available as part of the `M4DApplication` and can be printed from your terminal with: